radar-auth/src/main/java/org/radarcns/auth/authentication/TokenValidator.java

```diff
@@ -169,47 +176,49 @@ private JWTVerifier loadVerifier() throws TokenValidationException {
             lastFetch = Instant.now();
         }
 
-        RSAPublicKey publicKey;
-        if (config.getPublicKey() == null) {
-            publicKey = publicKeyFromServer();
-        } else {
-            publicKey = config.getPublicKey();
+        List<PublicKey> publicKeys = new LinkedList<>();
+        if (config.getPublicKeyEndpoints() != null) {
+            publicKeys.addAll(config.getPublicKeyEndpoints().stream()
+                    .map(this::publicKeyFromServer).collect(Collectors.toList()));
         }
-        Algorithm alg = Algorithm.RSA256(publicKey, null);
-        // we successfully fetched the public key, reset the timer
-        return JWT.require(alg)
+        if (config.getPublicKeys() != null) {
+            publicKeys.addAll(config.getPublicKeys());
+        }
+
+        // Create a verifier for each public key we have in our config
+        return publicKeys.stream().map(key -> JWT.require(publicKeyToAlgorithm(key))
                 .withAudience(config.getResourceName())
-                .build();
+                .build())
+                .collect(Collectors.toList());
     }
 
-    private RSAPublicKey publicKeyFromServer() throws TokenValidationException {
-        log.info("Getting the JWT public key at " + config.getPublicKeyEndpoint());
-
+    private PublicKey publicKeyFromServer(URI serverUri) throws TokenValidationException {
+        log.info("Getting the JWT public key at " + serverUri);
         try {
-            URLConnection connection =  config.getPublicKeyEndpoint().toURL().openConnection();
+            URLConnection connection =  serverUri.toURL().openConnection();
             connection.setRequestProperty("Accept", "application/json");
             try (InputStream inputStream = connection.getInputStream()) {
                 ObjectMapper mapper = new ObjectMapper();
                 JsonNode publicKeyInfo = mapper.readTree(inputStream);
-
-                // We expect RSA algorithm, and deny to trust the public key otherwise, see also
+                // We expect RSA or ECDSA algorithm, and deny to trust the public key otherwise
                 // https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/
-                if (!publicKeyInfo.get("alg").asText().equals("SHA256withRSA")) {
-                    throw new TokenValidationException("The identity server reported the following "
-                        + "signing algorithm: " + publicKeyInfo.get("alg")
-                        + ". Expected SHA256withRSA.");
+                if (publicKeyInfo.get("alg").asText().equals("SHA256withRSA")) {
```

Comment:
As an alternative, you could farm the public key parsing and algorithm generation to a separate class. You could load the algorithms in the constructor.
```java
interface TokenValidationAlgorithm {
    String getJwtAlgorithm(); // SHA256withRSA
    String keyHeader(); // "-----BEGIN PUBLIC KEY-----"
    Algorithm getAlgorithm(String keyValue); // Algorithm.RSA256(parseKey(keyValue), null);
}
```
Then don't parse the public keys in the configuration, and directly load the public key + algorithm from string.